vlasiator.cpp

```diff
@@ -485,6 +485,13 @@ int main(int argn,char* args[]) {
       *project
    );
    
+   // There are projects that have non-uniform and non-zero perturbed B, e.g. Magnetosphere with dipole type 4.
+   // For inflow cells (e.g. maxwellian), we cannot take a FSgrid perturbed B value from the templateCell,
+   // because we need a copy of the value from initialization in both perBGrid and perBDt2Grid and it isn't
+   // touched as we are in boundary cells for components that aren't solved. We do a straight full copy instead
+   // of looping and detecting boundary types here.
+   perBDt2Grid = perBGrid;
```

Comment:
So with time-varying inflow, are we keeping this as is or should we have perBDt2Grid obtain the potentially different values from the time-variation?

Reply:
Hmm. You're correct, perBDt2 needs to also get the correct values at the inflow. Hmm.

Reply:
Sorry... TBH this now sounds like a massive can of worms – we don't want to replace the dt/2 B entries in the RK algorithm with these interpolated values, do we? And if we do it'll be a massive PITC. And that wasn't done by the OP so I'd leave this as is, maybe with a new issue for our archive and a request for @henry2004y's opinion on this? :)

Reply:
Well this is one of those things that the *correct* way would be to set up boundaries with the absolutely correct time indeed, to retain restart bit-identicality. However, the value of dt/2 should in most use scenarios be quite small with respect to the timescales of the time-varying inflow.

Reply:
On the other hand, at restarts we anyway only read one value of perb and one for E, not the _dt2 values. So...

Reply:
So I think I am wrong on this one, the half-timestep values are computed in the solver so we're good here, and shall live with non-half dt values for moments.

Reply:
Indeed this will now update the state of the BC pre-translation, and use that value for both for the perB and the perBDt2 fields. But it's slow evolution so should be fine, and works the same for restarts and non-restart runs.